torch/onnx/_internal/exporter/_building.py

```diff
@@ -0,0 +1,516 @@
+            and all(isinstance(val, ir.Value) for val in arg)
+        ):
+            # Skip the sequence of ir.Value. This is a variadic input or a Sequence input
+            # NOTE: Variadic operators like Max can be called with mixed ir.Value and Python constants
+            # like `Max(0, ir.Value())`
+            # We need to convert the Python constants to Constant nodes
+            # NOTE: Important to check that arg is not empty because we need to treat it as list[int] or list[float]
+            continue
+            # if param.variadic:
+            #     # FXIME: Handle variadic inputs and sequence inputs differently
+            #     raise NotImplementedError
+            # TODO: Find a way to recursively build constants. Maybe extract the logic out.
+            # FIXME: I am here
+
+        assert isinstance(
+            param, _schemas.Parameter
+        ), f"Expected Parameter, got {type(param)}"
+
+        if param.type_constraint in type_binding:
+            # A known dtype is available
+            dtype = type_binding[param.type_constraint].dtype
+        elif len(param.type_constraint.allowed_types) == 1:
+            # Only one type is allowed
+            dtype = next(iter(param.type_constraint.allowed_types)).dtype
+        else:
+            # No dtype information available. Infer from the Python constant
+            if isinstance(arg, bool):
+                dtype = ir.DataType.BOOL
+            elif isinstance(arg, float):
+                dtype = ir.DataType.FLOAT
+            elif isinstance(arg, int):
+                dtype = ir.DataType.INT64
+            elif isinstance(arg, str):
+                dtype = ir.DataType.STRING
+            elif isinstance(arg, (tuple, list)) and all(
+                isinstance(val, int) for val in arg
+            ):
+                dtype = ir.DataType.INT64
+            elif isinstance(arg, (tuple, list)) and any(
+                isinstance(val, float) for val in arg
+            ):
+                # NOTE: if any float is present, the dtype is float
+                dtype = ir.DataType.FLOAT
+            elif isinstance(arg, (ir.Tensor, ir.TensorProtocol)):
+                dtype = arg.dtype
+            elif arg is None:
+                dtype = ir.DataType.UNDEFINED
+            else:
+                raise TypeError(
+                    f"Constant input '{arg}' of type '{type(arg)}' is not supported"
+                )
+
+        if arg is None:
+            constant_value = None
+        elif not isinstance(arg, (ir.Tensor, ir.TensorProtocol)):
+            # Deduplicate the constants
+            if isinstance(arg, (tuple, list)):
+                # Make the arg hashable
+                arg = tuple(arg)  # noqa: PLW2901
+            constant_value = constant_farm.get((arg, dtype))  # type: ignore[arg-type]
+            if constant_value is None:
+                constant_tensor = ir.tensor(value=arg, dtype=dtype)  # type: ignore[arg-type]
```

Comment:
Is this a typo? it.Tensor?

Reply:
This is expected. ir.tensor will create a tensor.